### PR TITLE
fixes #46, fixes #65

### LINKED ISSUE #46: ivar trim doesn't fail when providing an improper BED file
**Describe the bug**
Previously in version 1.2, if an improper BED file was provided, then `ivar trim` would fail. However in 1.2.2, if an improper BED file is provided, then `ivar trim` won't fail, it will simply skip trimming, but still output a bam file, as if no `-b` option had been provided, potentially causing the error to go unnoticed. Related to https://github.com/andersen-lab/ivar/issues/35

**To Reproduce**
1. Download the BED file from https://github.com/artic-network/artic-ncov2019/blob/master/primer_schemes/nCoV-2019/V3/nCoV-2019.bed. The 5th column is not an integer so `ivar trim` will consider it invalid.
2. Run `ivar trim` using this bed file. `ivar trim` will complain about the BED file format:
```
iVar uses the standard 6 column BED format as defined here - https://genome.ucsc.edu/FAQ/FAQformat.html#format1.
It requires the following columns delimited by a tab: chrom, chromStart, chromEnd, name, score, strand
```
But then proceed anyways, resulting in no reads being trimmed:
```
Trimmed primers from 0% (0) of reads.
0.73% (10132) of reads were quality trimmed below the minimum length of 30 bp and were not writen to file.
99.27% (1376097) of reads started outside of primer regions. Since the -e flag was given, these reads were written to file.
```

### LINKED ISSUE #65: TOTAL_DP is inconsistent with -m parameter
**Describe the bug**

Occasionally variants are called where TOTAL_DP is less than the user-provided depth threshold (`-m`) (for example with `-m 10` we observe variants with `TOTAL_DP` as low as 3). It appears that ivar checks depth based on the raw mpileup output, prior to any filtering (for example by quality score `-q`). This may lead to low quality variant/consensus calls when most of the reads don't pass the quality filter.

**Expected behavior**

I would find it more intuitive, and possibly lead to better results, if the depth check was applied after any read filtering.

### PR DESCRIPTION
**Pull request by AlaaALatif**
Minor code modifications to resolve issues #46 and #65:
`ivar trim` fails if bed file is provided but no primers identified.
`ivar trim` gives warning if bed file contains the score column with incorrect format, but continues trimming normally.
`ivar variants` applies minimum depth check after read filtering is carried out.

### CODE REVIEW

File: src/primer_bed.cpp [hunk 1]
```diff
@@ -151,13 +152,16 @@ std::vector<primer> populate_from_file(std::string path){
     primers.push_back(p);
     indice++;
   }
-  std::cout << "Found " << primers.size() << " primers in BED file" << std::endl;
-  if(primers.size() == 0){
+
+if(primers.size() == 0){
```
Comment (gkarthik) [inline src/primer_bed.cpp:156]: Not sure why this check is necessary here?

File: src/primer_bed.cpp [hunk 2]
```diff
@@ -86,6 +86,7 @@ void print_bed_format(){
   std::cout << "It requires the following columns delimited by a tab: chrom, chromStart, chromEnd, name, score, strand" << std::endl;
 }
 
+
```
Comment (gkarthik) [inline src/primer_bed.cpp:89]: Remove extra line here

File: src/primer_bed.cpp [hunk 3]
```diff
@@ -127,9 +128,9 @@ std::vector<primer> populate_from_file(std::string path){
 	if(std::all_of(cell.begin(), cell.end(), ::isdigit)) {
 	  p.set_score(stoi(cell));
 	} else {
-	  print_bed_format();
-	  primers.clear();
-	  return primers;
+	  print_bed_format();  // score is missing, send warning but continue populating
```
Comment (gkarthik) [inline src/primer_bed.cpp:131]: Warning seems too wordy. The following should suffice.
`WARNING: The provided BED file did not have the expected score column but iVar will continue trimming.`

File: src/trim_primer_quality.cpp [hunk 4]
```diff
@@ -324,12 +324,14 @@ int trim_bam_qual_primer(std::string bam, std::string bed, std::string bam_out,
...
     primers = populate_from_file(bed);
+    if(primers.size() == 0){
+      std::cout << "Exiting." << std::endl;
+      return -1;
+    }
   }
-  // if(primers.size() == 0){
-  //   return 0;
-  // }
+
```
Comment (gkarthik) [inline src/trim_primer_quality.cpp:332]: Remove extra lines

File: src/trim_primer_quality.cpp [hunk 5]
```diff
@@ -324,12 +324,14 @@ int trim_bam_qual_primer(std::string bam, std::string bed, std::string bam_out,
...
+      return -1;
+    }
   }
-  // if(primers.size() == 0){
-  //   return 0;
-  // }
+  
   if(bam.empty()){
-    std::cout << "Bam file in empty." << std::endl;
+    std::cout << "Bam file is empty." << std::endl;
```
Comment (gkarthik) [inline src/trim_primer_quality.cpp:334]: 😅 I've left quite a few typos here!!

Comment (AlaaALatif) [reply src/trim_primer_quality.cpp:334]: ya that's expected. I'll keep fixing the ones that I run into!

**Comment by jts** [comment]
Hi @gkarthik and @AlaaALatif,

Can I bug you to merge this PR and release a new version on conda please? We're preparing a new version of our pipeline and would like to incorporate these changes.

Thanks!
Jared

**Comment by gkarthik** [comment]
For sure! We will try pushing a release today!

**Comment by jts** [comment]
thanks, I appreciate it
